bindings/ocaml/EverCrypt.ml

```diff
@@ -0,0 +1,282 @@
+open Ctypes
+open Unsigned
+
+open SharedDefs
+open SharedFunctors
+module C = CBytes
+
+module Hacl_Spec = Hacl_Spec_bindings.Bindings(Hacl_Spec_stubs)
+
+module EverCrypt_AutoConfig2 = EverCrypt_AutoConfig2_bindings.Bindings(EverCrypt_AutoConfig2_stubs)
+module EverCrypt_AEAD = EverCrypt_AEAD_bindings.Bindings(EverCrypt_AEAD_stubs)
+module EverCrypt_Chacha20Poly1305 = EverCrypt_Chacha20Poly1305_bindings.Bindings(EverCrypt_Chacha20Poly1305_stubs)
+module EverCrypt_Curve25519 = EverCrypt_Curve25519_bindings.Bindings(EverCrypt_Curve25519_stubs)
+module EverCrypt_Hash = EverCrypt_Hash_bindings.Bindings(EverCrypt_Hash_stubs)
+module EverCrypt_HMAC = EverCrypt_HMAC_bindings.Bindings(EverCrypt_HMAC_stubs)
+module EverCrypt_Poly1305 = EverCrypt_Poly1305_bindings.Bindings(EverCrypt_Poly1305_stubs)
+module EverCrypt_HKDF = EverCrypt_HKDF_bindings.Bindings(EverCrypt_HKDF_stubs)
+module EverCrypt_DRBG = EverCrypt_DRBG_bindings.Bindings(EverCrypt_DRBG_stubs)
+module EverCrypt_Ed25519 = EverCrypt_Ed25519_bindings.Bindings(EverCrypt_Ed25519_stubs)
+
+
+module AutoConfig2 = struct
+  open EverCrypt_AutoConfig2
+  let init () = everCrypt_AutoConfig2_init ()
+  let has_shaext () = everCrypt_AutoConfig2_has_shaext ()
+  let has_aesni () = everCrypt_AutoConfig2_has_aesni ()
+  let has_pclmulqdq () = everCrypt_AutoConfig2_has_pclmulqdq ()
+  let has_avx2 () = everCrypt_AutoConfig2_has_avx2 ()
+  let has_avx () = everCrypt_AutoConfig2_has_avx ()
+  let has_bmi2 () = everCrypt_AutoConfig2_has_bmi2 ()
+  let has_adx () = everCrypt_AutoConfig2_has_adx ()
+  let has_sse () = everCrypt_AutoConfig2_has_sse ()
+  let has_movbe () = everCrypt_AutoConfig2_has_movbe ()
+  let has_rdrand () = everCrypt_AutoConfig2_has_rdrand ()
+end
+
+module Error = struct
+  type error_code =
+    | UnsupportedAlgorithm
+    | InvalidKey
+    | AuthenticationFailure
+    | InvalidIVLength
+    | DecodeError
+  type 'a result =
+    | Success of 'a
+    | Error of error_code
+  let error n =
+    let err = match n with
+      | 1 -> UnsupportedAlgorithm
+      | 2 -> InvalidKey
+      | 3 -> AuthenticationFailure
+      | 4 -> InvalidIVLength
+      | 5 -> DecodeError
+      | _ -> failwith "Impossible"
+    in
+    Error err
+  let get_result r = match UInt8.to_int r with
+    | 0 -> Success ()
+    | n -> error n
+end
+
+module AEAD = struct
+  open Error
+  open Hacl_Spec
+  open EverCrypt_AEAD
```

Comment:
a function to free the key is missing here but I'm not sure it's even in the .fst file